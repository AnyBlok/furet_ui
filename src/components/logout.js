import axios from 'axios';
import { defineComponent } from './factory';

defineComponent('furet-ui-appbar-user-dropmenu', {
  template: `
    <div class="navbar-item has-dropdown is-hoverable">
      <a class="navbar-link">
        {{ userName }}
      </a>
      <div class="navbar-dropdown">
        <div class="navbar-item">
         <a v-on:click="logOut" class="button is-primary is-inverted is-fullwidth">
           {{ $t('components.logout.button') }}
         </a>
        </div>
      </div>
    </div>
  `,
  prototype: {
    computed: {
      userName() {
        return this.$store.state.global.userName;
      },
    },
    methods: {
      logOut() {
        axios.post('/furet-ui/logout')
          .then((result) => {
            this.$dispatchAll(result.data);
          })
      },
    },
  },
});
