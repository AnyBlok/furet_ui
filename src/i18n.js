/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueI18n from 'vue-i18n';
Vue.use(VueI18n);


export const i18nConf = {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
            fields: {
                common: {
                    required: 'this field is required',
                },
                json: {
                    invalid: 'Invalid JSON format',
                },
                file: {
                    upload: 'Upload',
                    download: 'Download',
                    delete: 'Delete',
                },
            },
        },
    },
};
export const i18n = new VueI18n(i18nConf);
