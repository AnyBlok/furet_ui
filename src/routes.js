/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueRouter from 'vue-router';
Vue.use(VueRouter);
import {json_post_dispatch_all} from './server-call';


const CustomeView = {
    template: '<furet-ui-custom-view v-bind:viewName="viewName" />',
    props: ['viewName'],
    beforeRouteEnter: (to, from, next) => {
        json_post_dispatch_all('/custom/view/' + to.params.viewName, {});
        next();
    },
    beforeRouteUpdate: (to, from, next) => {
        if (to.params.viewName != from.params.viewName) {
            json_post_dispatch_all('/custom/view/' + to.params.viewName, {});
        }
        next();
    },
}

const Action = {
    template: '<router-view></router-view>',
    beforeRouteEnter: (to, from, next) => {
        json_post_dispatch_all('/action/' + to.params.actionId, to.params);
        next();
    },
    beforeRouteUpdate: (to, from, next) => {
        if (to.params.actionId != from.params.actionId)
            json_post_dispatch_all('/action/' + to.params.actionId, to.params);
        next();
    },
};

const View = {
    template: `<furet-ui-view
                    v-bind:spaceId="spaceId"
                    v-bind:menuId="menuId"
                    v-bind:actionId="actionId"
                    v-bind:viewId="viewId"
                    v-bind:dataId="dataId"
                    v-bind:mode="mode"
                />`,
    props: ['spaceId', 'menuId', 'actionId', 'viewId', 'dataId', 'mode'],
    beforeRouteEnter: (to, from, next) => {
        json_post_dispatch_all('/view/' + to.params.viewId, to.params);
        next();
    },
    beforeRouteUpdate: (to, from, next) => {
        if (to.params.viewId != from.params.viewId)
            json_post_dispatch_all('/view/' + to.params.viewId, to.params);
        next();
    },
};


export const router = new VueRouter({
    routes: [
        {
            path: '/custom/view/:viewName',
            name: 'custom_view',
            props: true,
            component: CustomeView,
        },
        {
            path: '/space/:spaceId',
            name: 'space',
            props: true,
            component: {
                template: `<furet-ui-space
                                v-bind:spaceId="spaceId"
                                v-bind:menuId="menuId"
                                v-bind:actionId="actionId"
                                v-bind:viewId="viewId"
                            />`,
                props: ['spaceId', 'menuId', 'actionId', 'viewId'],
                beforeRouteEnter: (to, from, next) => {
                    json_post_dispatch_all('/space/' + to.params.spaceId, to.params);
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    if (to.params.spaceId != from.params.spaceId) json_post_dispatch_all('/space/' + to.params.spaceId, to.params);
                    next();
                },
            },
            children: [
                {
                    path: 'menu/:menuId',
                    name: 'space_menu',
                    component: {
                        template: '<router-view></router-view>'
                    },
                    children: [
                        {
                            path: 'action/:actionId',
                            name: 'space_menu_action',
                            component: Action,
                            children: [
                                {
                                    path: 'view/:viewId',
                                    name: 'space_menu_action_view',
                                    props: true,
                                    component: View,
                                    children: [
                                        {
                                            path: 'data/:dataId/mode/:mode',
                                            name: 'space_menu_action_view_dataId',
                                            component: {template: '<div />'},  // never called
                                        },
                                    ],
                                },
                                {
                                    path: 'custom/view/:viewName',
                                    name: 'space_menu_action_custom_view',
                                    props: true,
                                    component: CustomeView,
                                },
                            ],
                        },
                    ],
                },
                {
                    path: 'action/:actionId',
                    name: 'space_action',
                    component: Action,
                    children: [
                        {
                            path: 'view/:viewId',
                            name: 'space_action_view',
                            props: true,
                            component: View,
                            children: [
                                {
                                    path: 'data/:dataId/mode/:mode',
                                    name: 'space_action_view_dataId',
                                    component: {template: '<div />'},  // never called
                                },
                            ],
                        },
                        {
                            path: 'custom/view/:viewName',
                            name: 'space_action_custom_view',
                            props: true,
                            component: CustomeView,
                        },
                    ],
                },
            ]
        },
    ],
});

export default router;
