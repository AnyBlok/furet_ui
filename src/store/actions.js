/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export const UNITEST_CLEAR = ({ commit }) => {
    commit('CLEAR_CLIENT');
    commit('CLEAR_DATA');
    commit('CLEAR_GLOBAL');
    commit('CLEAR_LEFT_MENU');
    commit('CLEAR_RIGHT_MENU');
}
