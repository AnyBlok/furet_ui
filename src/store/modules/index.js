/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import global from './global';
import client from './client';
import rightmenu from './right-menu';
import leftmenu from './left-menu';
import data from './data';

export default {
    global,
    client,
    rightmenu,
    leftmenu,
    data,
}
