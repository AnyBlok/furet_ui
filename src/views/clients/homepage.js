/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';

export const Homepage = Vue.component('furet-ui-homepage', {
    template: `
        <section class="section content">
            <div class="columns is-mobile">
                <div class="column is-half is-offset-one-quarter">
                    <h1>{{ $t('views.clients.homepage.title') }}</h1>
                    <p>{{ $t('views.clients.homepage.description') }}</p>
                </div>
            </div>
        </section>`,
});
