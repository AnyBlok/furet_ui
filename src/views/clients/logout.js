/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import plugin from '../../plugin';
import {json_post_dispatch_all} from '../../server-call';

export const Logout = ({router}) => {
    router.push('/');
    json_post_dispatch_all('/client/logout', {});
};

plugin.set(['views', 'type', 'client'], {Logout: {function: Logout}})
