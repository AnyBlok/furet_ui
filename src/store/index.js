/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import _ from 'underscore';
import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import modules from './modules';
import {i18n} from '../i18n';
import {router} from '../routes';
import moment from 'moment';

Vue.use(Vuex)
const debug = process.env.NODE_ENV !== 'production'

export const storeDef = {
    actions,
    getters,
    modules,
    strict: debug,
};

export const store = new Vuex.Store(storeDef);

export const dispatchAll = (datas) => {
    _.each(datas, data => {
        if (data.type) {
            switch (data.type) {
                case 'UPDATE_LOCALES':
                    _.each(data.locales, locale => {
                        i18n.setLocaleMessage(locale.locale, locale.messages)
                    });
                    break
                case 'SET_LOCALE':
                    i18n.locale = data.locale;
                    moment.locale(data.locale);
                    break
                case 'UPDATE_ROUTE':
                    if (data.name) router.push({name: data.name, params: data.params});
                    else if (data.path) router.push({path: data.path});
                    break;
                default:
                    const type = data.type;
                    delete data.type;
                    const process = data.process || 'commit';
                    store[process](type, data);
            }
        }
    });
}

export default store;
