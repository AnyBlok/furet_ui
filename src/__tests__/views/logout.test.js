/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import {store} from '../../store';
import {router} from '../../routes';
import {i18n} from '../../i18n';
import {Logout} from '../../views/clients/logout';
jest.mock('../../server-call')

describe('App component', () => {
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render App with default value from store', () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push('/space/1/menu/2/action/3/view/4');
        expect(vm.$route.path).toBe('/space/1/menu/2/action/3/view/4');
        Logout({router})
        expect(vm.$route.path).toBe('/');
    });
});
