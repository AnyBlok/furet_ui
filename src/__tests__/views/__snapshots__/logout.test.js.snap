// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login 1`] = `null`;
